src/Controls/src/Core/Handlers/Items2/iOS/CarouselViewController2.cs

```diff
@@ -193,6 +193,11 @@ internal int GetIndexFromIndexPath(NSIndexPath indexPath)
 			return indexPath.Row;
 		}
 
+		bool IsValidIndexPath(NSIndexPath indexPath)
+		{
+			return !(indexPath.Row < 0 || indexPath.Row >= CollectionView.NumberOfItemsInSection(indexPath.Section));
```

Comment:
Could we expand the validations?
Should return false `if (indexPath == null)` or `if (indexPath.Section < 0)`